### PR TITLE
added metadata to examples/plotting/polar.py

### PR DESCRIPTION
**Pull request by akshatvishu**
Metadata added for examples/plotting/polar.py
- [ ] issues: fixes #11765

### CODE REVIEW

File: examples/plotting/polar.py [hunk 1]
```diff
@@ -1,3 +1,10 @@
+'''This example shows how to use polar coordinates to generate a gird of polar plots of rose/rohdonea curves.
```
Comment (mosc9575) [inline examples/plotting/polar.py]:
```suggestion
'''This example shows how to use polar coordinates to generate a grid of polar plots of rose/rohdonea curves.
```

File: examples/plotting/polar.py [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+
+.. bokeh-example-metadata::
+    :apis: bokeh.colors.named, bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.Plot, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
+    :keywords: polar coordinates, rose curves, rohdonea curves
```
Comment (mosc9575) [inline examples/plotting/polar.py]:
```suggestion
    :apis: bokeh.colors.named, bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.Plot, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
    :refs: :ref:`ug_topics_contour` > :ref:`ug_topics_contour_polar` 
    :keywords: polar coordinates, rose curves, rohdonea curves
```

**Review by mosc9575 — COMMENTED** [review]
I found one typo and I think it would be usefull to link the one polar coordinates example from the userguide.

**Comment by akshatvishu** [comment]
Thank you for your comments, @mosc9575 ! I appreciate the feedback. I realize I made mistakes with the references, and I had difficulty finding them using these instructions (https://github.com/bokeh/bokeh/issues/11765). If you could please tell me how to navigate and find references, I would greatly appreciate it.

**Comment by mosc9575** [comment]
> Thank you for your comments, @mosc9575 ! I appreciate the feedback. I realize I made mistakes with the references, and I had difficulty finding them using these instructions (#11765). If you could please tell me how to navigate and find references, I would greatly appreciate it.

It is not so easy to find them, but all links are someweher in a ".rst"-file in the folder `docs/bokeh/source/docs/user_guide`. In your case it was in `ocs/bokeh/source/docs/user_guide/topics/contour.rst`. In the "raw" version of the ".rst"-file there are some lines like

```
.. _ug_topics_contour:
```
 
starting with `.. _`. The text afterwards is your reference.

**Comment by akshatvishu** [comment]
> It is not so easy to find them, but all links are someweher in a ".rst"-file in the folder `docs/bokeh/source/docs/user_guide`. In your case it was in `ocs/bokeh/source/docs/user_guide/topics/contour.rst`. In the "raw" version of the ".rst"-file there are some lines like

So, there is no easy-way ! Have to go through all .rst files to see if it contains the link ?( (๑•́ㅿ•̀๑) ᔆᵒʳʳᵞ, I made you do all the hard work for this time! )

Also, I made all the suggested changes that was proposed! Hopefully, this will get merge without any issues.

**Comment by mosc9575** [comment]
Don't feel sorry.  It's not that bad. You can always use the search function on the official docs or inside your repository. This gives you some hints where so find a good section or header.

File: examples/plotting/polar.py [hunk 3]
```diff
@@ -1,3 +1,11 @@
+'''This example shows how to use polar coordinates to generate a grid of polar plots of rose/rohdonea curves.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.colors.named, bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.Plot, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
+    :refs: :ref:`ug_topics_contour` > :ref:`ug_topics_contour_polar`
```
Comment (mosc9575) [inline examples/plotting/polar.py]:
```suggestion
    :refs: :ref:`ug_topics_contour` > :ref:`ug_topics_contour_polar`
```

Here is a trailling whitespace. This is the reason why one test failed.

File: examples/plotting/polar.py [hunk 4]
```diff
@@ -1,3 +1,11 @@
+'''This example shows how to use polar coordinates to generate a grid of polar plots of rose/rohdonea curves.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.colors.named, bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.Plot, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
```
Comment (mosc9575) [inline examples/plotting/polar.py]:
```suggestion
    :apis: bokeh.colors.named, bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.Plot, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot # noqa: E501
```
This line violates the max line length of 165 letters. To silence this warning, you can add `# noqa: E501` to the end of the line or of the whole code block.

But I would suggest to remove 2 elements
```suggestion
    :apis: bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
```

Comment (akshatvishu) [reply examples/plotting/polar.py]: > This line violates the max line length of 165 letters. To silence this warning, you can add `# noqa: E501` to the end of the line or of the whole code block.
> 
> But I would suggest to remove 2 elements

Thanks!, as you've suggested, I've removed these two elements + whitespace from earlier !

**Comment by bryevdv** [comment]
> So, there is no easy-way ! Have to go through all .rst files 

@akshatvishu FYI I use a code grep tool like `ripgrep` for this. E.g. if I want to search for "contour" references, I might do something like:
```
base ❯ rg "_contour"
source/docs/reference/plotting/contour.rst
7:.. autofunction:: bokeh.plotting.contour.from_contour

source/docs/user_guide/topics/contour.rst
1:.. _ug_topics_contour:
10:.. _ug_topics_contour_simple:
43:.. _ug_topics_contour_polar:
77:.. _ug_topics_contour_animated:

source/docs/user_guide/topics.rst
11::ref:`ug_topics_contour`
```

File: examples/plotting/polar.py [hunk 5]
```diff
@@ -1,3 +1,12 @@
+'''This example shows hot to use polar coordinates to generate a
+grid of polar plots of rose/rohdonea curves.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.io.show, bokeh.models.ColumnDataSource, bokeh.models.PolarTransform, bokeh.plotting.figure, bokeh.plotting.figure.line, bokeh.plotting.gridplot
+    :refs: :ref:`ug_topics_contour` > :ref:`ug_topics_contour_polar`
```
Comment (bryevdv) [inline examples/plotting/polar.py]: We stopped doing the "breadcrumbs" style with references, that's just too hard to maintain. Only need to add the "last" reference in the chain:
```suggestion
    :refs: :ref:`ug_topics_contour_polar`
```

File: examples/plotting/polar.py [hunk 6]
```diff
@@ -1,3 +1,12 @@
...
+    :keywords: polar coordinates, rose curves, rohdonea curves
```
Comment (bryevdv) [inline examples/plotting/polar.py:7]: I've never heard the term "rohdonea" before. However, if it is common in some places that I am unaware, then this suggestion can be ignored. 
```suggestion
    :keywords: polar coordinates, rose curves
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Thanks @akshatvishu just a couple of final suggestions

File: examples/plotting/polar.py (hunk 6 above)
Comment (akshatvishu) [reply examples/plotting/polar.py:7]: > I've never heard the term "rohdonea" before. However, if it is common in some places that I am unaware, then this suggestion can be ignored.

Me neither but I saw the [wiki](https://en.wikipedia.org/wiki/Rose_(mathematics)) article on it  and took it from there.

File: examples/plotting/polar.py (hunk 5 above)
Comment (akshatvishu) [reply examples/plotting/polar.py:6]: > We stopped doing the "breadcrumbs" style with references, that's just too hard to maintain. Only need to add the "last" reference in the chain:

File: examples/plotting/polar.py (hunk 6 above)
Comment (akshatvishu) [reply examples/plotting/polar.py:7]: > I've never heard the term "rohdonea" before. However, if it is common in some places that I am unaware, then this suggestion can be ignored.

I took it from the  [wiki](https://en.wikipedia.org/wiki/Rose_(mathematics)) article describing rose curves:
'_In [mathematics](https://en.wikipedia.org/wiki/Mathematics), a rose or rhodonea curve is a [sinusoid](https://en.wikipedia.org/wiki/Sine_wave) specified by either the [cosine](https://en.wikipedia.org/wiki/Cosine) or [sine](https://en.wikipedia.org/wiki/Sine) functions with no [phase angle](https://en.wikipedia.org/wiki/Phase_(waves)) that is plotted in [polar coordinates](https://en.wikipedia.org/wiki/Polar_coordinates)._'

File: examples/plotting/polar.py (hunk 5 above)
Comment (akshatvishu) [reply examples/plotting/polar.py]: > Only need to add the "last" reference in the chain

sure!

**Comment by akshatvishu** [comment]
> > So, there is no easy-way ! Have to go through all .rst files
> 
> @akshatvishu FYI I use a code grep tool like `ripgrep` for this. E.g. if I want to search for "contour" references, I might do something like:
> 
> ```
> base ❯ rg "_contour"
> source/docs/reference/plotting/contour.rst
> 7:.. autofunction:: bokeh.plotting.contour.from_contour
> 
> source/docs/user_guide/topics/contour.rst
> 1:.. _ug_topics_contour:
> 10:.. _ug_topics_contour_simple:
> 43:.. _ug_topics_contour_polar:
> 77:.. _ug_topics_contour_animated:
> 
> source/docs/user_guide/topics.rst
> 11::ref:`ug_topics_contour`
> ```

Thank you so much!

File: examples/plotting/polar.py [hunk 7]
```diff
@@ -1,3 +1,12 @@
+'''This example shows hot to use polar coordinates to generate a
```
Comment (bryevdv) [inline examples/plotting/polar.py]: typo
```suggestion
'''This example shows how to use polar coordinates to generate a
```

File: examples/plotting/polar.py (hunk 6 above)
Comment (bryevdv) [reply examples/plotting/polar.py:7]: It seems a bit obscure I would suggest removing it

File: examples/plotting/polar.py (hunk 7 above)
Comment (akshatvishu) [reply examples/plotting/polar.py]: Sorry!  I just fixed it!

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks  for the PR @akshatvishu !

**Comment by akshatvishu** [comment]
> Thanks for the PR @akshatvishu !
You're welcome, glad to contribute!

**Pull request merged**